File: solutions/c/resistor-color/12/resistor_color.c
#include "resistor_color.h"

int color_code(resistor_band_t band) {
    return band;
}

resistor_band_t *colors() {
    static resistor_band_t values[10];
    for (resistor_band_t i = BLACK; i <= WHITE; i++) {
        values[i] = i;
    }
    return values;
}

File: solutions/java/resistor-color/2/src/main/java/ResistorColor.java
import java.util.*;

class ResistorColor {
    Map <String, Integer> map = new LinkedHashMap<>() {{
        put("black", 0);
        put("brown", 1);
        put("red", 2);
        put("orange", 3);
        put("yellow", 4);
        put("green", 5);
        put("blue", 6);
        put("violet", 7);
        put("grey", 8);
        put("white", 9);                                 
    }};
    int colorCode(String color) {
        return map.get(color);
    }

    String[] colors() {
        String [] colorsArray = (map.keySet().toArray(new String[0]));
        return colorsArray; 
    }
}
